### PR TITLE
[browser] Fix computing destination sub path and publish extension target path in Wasm SDK

### LINKED ISSUE #105393: [Wasm] Incorrect boot.json for extensions
The JSON generated when the build defines publish extensions is wrong.
image

The bit that fails is as follows:
image

We are expecting `resources.extensions.my-custom-extension["_bin/publish.extension.txt"]` to be populated, but we got "" instead.

### LINKED ISSUE #105399: [Wasm] Issue when defining assets for localized assemblies
The identities for the assets are duplicate when they should not.

image

### PR DESCRIPTION
**Pull request by maraf**
Regression introduced in https://github.com/dotnet/runtime/pull/103755. 
This code path is used rarely, we don't have a test coverage in runtime repo, tested with blazor test in SDK repo.

Fixes https://github.com/dotnet/runtime/issues/105399
Fixes https://github.com/dotnet/runtime/issues/105393

### CODE REVIEW

File: src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/AssetsComputingHelper.cs [hunk 1]
```diff
@@ -107,7 +107,7 @@ public static string GetCandidateRelativePath(ITaskItem candidate, bool fingerpr
         {
             fileName = Path.GetFileNameWithoutExtension(destinationSubPath);
             extension = Path.GetExtension(destinationSubPath);
-            subPath = destinationSubPath.Substring(fileName.Length + extension.Length);
+            subPath = destinationSubPath.Substring(0, destinationSubPath.Length - (fileName.Length + extension.Length)); ;
```
Comment (ilonatommy) [inline src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/AssetsComputingHelper.cs:110]: that's fine. Why not `Path.GetDirectoryName(destinationSubPath);`?

**Review by ilonatommy — APPROVED** [review]

File: src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/AssetsComputingHelper.cs (hunk 1 above)
Comment (maraf) [reply src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/AssetsComputingHelper.cs:110]: Sounds good!

**Comment by maraf** [comment]
/ba-g The only untracted failure is timeout unrelated to the changes here

**Pull request merged**